src/SkyCoords.jl

```diff
@@ -7,12 +7,74 @@ export AbstractSkyCoords,
        ICRSCoords,
        GalCoords,
        FK5Coords,
-       separation
-
-import Base: convert
+       separation,
+       str2rad
 
 include("types.jl")
 
+"""
+    str2rad(::AbstractString, force_ha=false)
+
+Parses strings that specify common astronomical coordinates into radians. A simplified version of the regex allowed for both
+
+* hour angle - `"xx [h] xx ['′m:] xx [\\\"″s]?"`
+* degree - `"xx [°d:] xx ['′m:] xx [\\\"″s]?"`
+
+So, you can see that this will parse an hour angle if the leading unit contains an `h` otherwise it will parse a degree if the leading unit contains a `°`, `d` or `:`. Also, you can forego the final unit, since it will be parsed automatically. The values (`xx`) will be parsed as `Float64`.
+
+If `force_ha` is `true` then the input will be parsed as an hour angle using a looser regex that allows the leading unit to be a `:`
+
+* loose hour angle - `"xx [h:] xx ['′m:] xx [\\\"″s]?"`
+
+# Examples
+```jldoctest
+julia> str2rad("12h52m64.300s")
+3.3731614843033575
+
+julia> str2rad("0°12'5\\\"") # Note that you have to escape the "
+0.003514899188044136
+
+julia> str2rad("-  10° 02 ′  10.885 ″") # Whitespace does not matter
+-0.17389837681291273
+
+julia> str2rad("12:0:0", true)
+3.141592653589793
+```
+"""
+function str2rad(input::AbstractString, force_ha=false) 
+    input = replace(strip(input), r"\s" => "")
+    ha_r = r"([+-]?\d+\.?\d*)h(\d+\.?\d*)['′m:](\d+\.?\d*)[\"″s]?"
+    ha_r_loose = r"([+-]?\d+\.?\d*)[h:](\d+\.?\d*)['′m:](\d+\.?\d*)[\"″s]?"
+    deg_r = r"([+-]?\d+\.?\d*)[°d:](\d+\.?\d*)['′m:](\d+\.?\d*)[\"″s]?"
+    if force_ha
+        m = match(ha_r_loose, input)
+        isnothing(m) && error("Could not parse $input as hour angle")
+        h, m, s = parse.(Float64, m.captures)
+        rad = hr2rad(h)
+        rad += hr2rad(m / 60)
+        rad += hr2rad(s / 3600)
+        return rad
+    elseif occursin(ha_r, input)
+        m = match(ha_r, input)
+        h, m, s = parse.(Float64, m.captures)
+        rad = hr2rad(h)
+        rad += hr2rad(m / 60)
+        rad += hr2rad(s / 3600)
+        return rad
+    elseif occursin(deg_r, input)
+        m = match(deg_r, input)
+        d, m, s = parse.(Float64, m.captures)
+        rad = deg2rad(d)
+        rad += deg2rad(m / 60)
+        rad += deg2rad(s / 3600)
+        return rad
+    else
+        error("Could not parse $input to sky coordinates")
+    end
+end
+
+@inline hr2rad(h::Number) = 2π/24 * h
```

Comment:
Fewer operations and preserves the same precision as `h`
```suggestion
@inline hr2rad(h::Number) = π * h / 12
```

E.g., with current definition:
```julia
julia> 1 - hr2rad(big(12)) / pi
1.096607112466052357390335435978052540497372671893099927562721861551176434983587e-16
````
with the proposed definition:
```julia
julia> 1 - hr2rad(big(12)) / pi
-1.727233711018888925077270372560079914223200072887256277004740694033718360632485e-77
```